docs/expectation_example.md

```diff
@@ -0,0 +1,120 @@
+## Motivation
```

Comment:
I have looked at this file after `expectation_example.py`, maybe it is worth it to just copy some of the commentary here to the runnable example (as comments)?

I haven't thought about it much, but there should be a nicer solution than duplicating the example code in a markdown file for the docs and a runnable Python file. Maybe a Jupyter notebook that can be rendered as a markdown file?